### DISCUSSION TITLE
Combining GIF

### DISCUSSION THREAD

**Discussion by Jay123210599** [ImageMagick/ImageMagick#7560] [Help]
### ImageMagick version

7.1.1-27

### Operating system, version and so on

Windows 11

### Description

How do I combine two gifs together, vertically and horizontally?

**Comment by fmw42** [comment]
See -append for vertical and +append for horizontal for two single frame gifs.
Are these animated gifs. If so a more complex method is needed. See https://imagemagick.org/Usage/anim_mods/#append

**Reply by Jay123210599** [reply]
Yes, they are animated.

**Reply by fmw42** [reply]
Do they have the same number of frames?

**Reply by fmw42** [reply]
Can you post your two animation gifs?

**Reply by Jay123210599** [reply]
https://drive.google.com/file/d/16VLZvRKC0UMGncBQzeoVOVxpQkcAAsH-/view?usp=sharing

**Comment by fmw42** [comment]
Your two animations do not have the same number of frames.  The only way to combine them is the drop frames from the one with more frames so that they have the same number of frames. Or include empty frames in the one with smaller number of frames.  What do you want to do here?

**Reply by fmw42** [reply]
Also please clarify that you want the animations side-by-side and not one on top of the other.

**Reply by Jay123210599** [reply]
Side-by-side, and also one on top of the other, if possible.

**Comment by fmw42** [comment]
Since the two animations have different number of frames and possibly frame delays, I am just taking the smallest number of frames and ignoring the frame delays.  One image has 35 and the other 29 frames. So I just take 29.  Also the frames sizes are the same at 570x907 as shown below for the first frame

```
magick identify ljk.gif
ljk.gif[0] GIF 570x907 570x907+0+0 8-bit sRGB 256c 0.000u 0:00.003
```

Doing left right appending, I need the height of the frames which is 907. I then append all vertically for both gifs and then the two vertical strips horizontally. Then I crop all the pairs vertically to get the horizontal appended animation.

Unix syntax:

```
magick \( ii.gif[0-29]  -coalesce +repage -append \) \
\( ljk.gif[0-29] -coalesce +repage -append \) \
+append  -crop x907 +repage \
-set delay 0  animation.gif

```

Windows syntax:

```
magick ( ii.gif[0-29]  -coalesce +repage -append ) ^
( ljk.gif[0-29] -coalesce +repage -append ) ^
+append  -crop x907 +repage ^
-set delay 0  animation.gif
```

Result:

[animation.gif.zip](https://github.com/user-attachments/files/16630465/animation.gif.zip)


For top bottom appending, swap the - and + on all the appends and change the crop to -crop 570x

**Reply by fmw42** [reply]
I told you how to make the change in the message above.  Can you not do that?

> For top bottom appending, swap the - and + on all the appends and change the crop to -crop 570x

Unix syntax:

```
magick \( ii.gif[0-29]  -coalesce +repage +append \) \
\( ljk.gif[0-29] -coalesce +repage +append \) \
-append  -crop 570x +repage \
-set delay 0  animation2.gif

```

Windows syntax:

```
magick ( ii.gif[0-29]  -coalesce +repage +append ) ^
( ljk.gif[0-29] -coalesce +repage +append ) ^
-append  -crop 570x +repage ^
-set delay 0  animation2.gif

```

[animation2.gif.zip](https://github.com/user-attachments/files/16630607/animation2.gif.zip)
